sdk/core/core/src/az_span.c

```diff
@@ -144,89 +144,57 @@ AZ_NODISCARD az_result az_span_find(az_span source, az_span target, az_span* out
+  const int32_t target_not_found = -1;
 
-  // This loop traverses `source` position by position (step 1.)
-  for (int i = 0; i < source_length; i++)
-  {
-    // This is the check done in step 1. above.
-    if (source_ptr[i] == target_ptr[0])
-    {
-      // The condition in step 2. has been satisfied.
-      int j;
-      // This is the loop defined in step 3.
-      // The loop must be broken if it reaches the ends of `target` (step 3.) OR `source` (step 5.).
-      for (j = 1; j < target_length && (i + j) < source_length; j++)
-      {
-        // Condition defined in step 5.
-        if (source_ptr[i + j] != target_ptr[j])
-        {
-            break;
-        }
-      }
- 
-      if (j == target_length)
-      {
-        // All bytes in `target` have been checked and matched the corresponding bytes in `source` (from the start point `i`),
-        // so this is indeed an instance of `target` in that position of `source` (step 4.).
-        // Here we create a span in `source` to map the occurrence of `target` in `source` (pardon the redundancy of words, done for explicit clarity).
-        *out_span = az_span_init(source_ptr + i, target_length, target_length);
-
-        return AZ_OK;
-      }
-    }
+  if (target_length == 0)
+  { 
+    return 0;
   }
-
-  // If the function hasn't returned before, all positions 
-  // of `source` have been evaluated but `target` could not be found.
-  return AZ_ERROR_ITEM_NOT_FOUND;
-}
-
-AZ_NODISCARD az_span az_span_token(az_span source, az_span delimiter, az_span* out_remainder)
-{
-  AZ_PRECONDITION_VALID_SPAN(delimiter, 1, false);
-  AZ_PRECONDITION_NOT_NULL(out_remainder);
-
-  if (az_span_is_content_equal(source, AZ_SPAN_NULL))
+  else if (source_length == 0) 
   {
-    return AZ_SPAN_NULL;
+    return target_not_found;
   }
   else
   {
     uint8_t* source_ptr = az_span_ptr(source);
-    int32_t source_length = az_span_length(source);
-    int32_t source_capacity = az_span_capacity(source);
+    uint8_t* target_ptr = az_span_ptr(target);
 
-    az_span instance;
-    if (az_span_find(source, delimiter, &instance) == AZ_OK)
+    // This loop traverses `source` position by position (step 1.)
+    for (int32_t i = 0; i < source_length; i++)
     {
-      uint8_t* instance_ptr = az_span_ptr(instance);
-      int32_t instance_length = az_span_length(instance);
-
-      *out_remainder = az_span_init(
-        instance_ptr + instance_length, 
-        (int32_t)(source_length - instance_length - (instance_ptr - source_ptr)),
-        (int32_t)(source_capacity - instance_length - (instance_ptr - source_ptr)));
-
-      return az_span_init(
-        source_ptr, 
-        (int32_t)(instance_ptr - source_ptr), 
-        (int32_t)(instance_ptr - source_ptr));
-    }
-    else
-    {
-      *out_remainder = AZ_SPAN_NULL;
+      // This is the check done in step 1. above.
+      if (source_ptr[i] == target_ptr[0])
+      {
+        // The condition in step 2. has been satisfied.
```

Comment:
what about calling az_span_is_equal(...) here, instead of below loop? 
We even have an az_span_is_equal_ignore_case()... (or something like that) in case we don't care about lower or upper case.

Reply:
I would say it's not as performant, because we only need to check one uint8_t at a time instead of the whole `target` every time with memcpy.